### PR TITLE
Dm 1920 qt launcher

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

A python/qt based EFU launcher.

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: utils/qtrunefu/qtdatagen.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    def get_selection(self):
+        return self._gen, self._data, self._efuip, self._throttle
+
+    # Override builtin on_accepted method for pressing OK button
+    def on_accepted(self):
+        self._gen = self.gencb.itemText(self.gencb.currentIndex())
+        self._data = self.datacb.itemText(self.datacb.currentIndex())
+        self._efuip = self.efuiple.text()
+        self._throttle = self.throttlele.text()
```
Comment (MortenHilkerSkaaning) [inline utils/qtrunefu/qtdatagen.py:147]: tja, jeg er ikke 100% vild med at det hele er lowercase ud i et. Det er self din stil, men jeg tror hurtigt at det kan vaere tricky, med laengere sammensaetninger. Vil camelCase ikke vaere ok? self.throttleLe, self.efuIpLe ?

File: utils/qtrunefu/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+# Qt based efu launcher
+This tool helps build a command line for launching efu.
+
+## qtefu.py
+The user speficies three directories: basedir, efudir and datadir. The latter two must
+be relative to basedir.
+
+The prgram then finds the relevant binary files (efu executable and detector libraries),
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply utils/qtrunefu/README.md:9]: will fix, and there are even more typos ....

File: utils/qtrunefu/qtdatagen.py (hunk 1 above)
Comment (mortenjc) [reply utils/qtrunefu/qtdatagen.py:147]: I will check with Matt Clarke, but I believe this is more or less the python coding style. If not I will make the necessary changes.

File: utils/qtrunefu/qtdatagen.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        self.options = {}
+        self.config = configparser.RawConfigParser()
+        self.config.read(os.path.join(expanduser("~"), ".efucfg"))
+        self._load('directories', 'basedir', expanduser("~"))
+        self._load('directories', 'efudir', '/essproj/event-formation-unit')
+        self._load('directories', 'datadir', '/ownCloud/DM/data/EFU_reference')
+        self._load('servers', 'grafana', '172.17.12.31')
+        self._load('servers', 'kafka', '172.17.5.38:9092')
+        self._load('servers', 'efu', '127.0.0.1')
+        self._load('datagen', 'throttle', '100')
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply utils/qtrunefu/qtdatagen.py:22]: nope, but the datagen is not readu for production yet. I have updated the README

File: utils/qtrunefu/qtdatagen.py (hunk 1 above)
Comment (mortenjc) [reply utils/qtrunefu/qtdatagen.py:147]: Matt replied:
PEP8 is the Python guidelines. https://pep8.org/
It is pretty much universally used.
In short, variable and function names are snake-case, e.g. detid or det_id, convert_to_tof() etc.
For the other aspects like spacing, indentation, etc. we tend to use the Black tool (https://pypi.org/project/black/) to automate that to conform to PEP8

**Pull request merged**